cuqi/utilities/_utilities.py

```diff
@@ -173,18 +163,46 @@ def approx_derivative(func, wrt, direction=None, epsilon=np.sqrt(np.finfo(np.flo
     # corresponding geometry par2fun map is not identity
     # (e.g. funvalues=paramters**2), because wrt entries 
     # are interpreted as function value.
-    if isinstance(wrt, CUQIarray) or isinstance(direction, CUQIarray):
-        raise NotImplementedError("approx_derivative is not implemented"+
-                                  "for inputs of type CUQIarray")
+
```

Comment:
Are the above comments still relevant? An error doesn't appear to be thrown anymore?

Reply:
Thank you for pointing this out. I thought about it for a bit and I think it needs more consideration, and I reverted the changes here to issue an error if the input is cuqi array. I added an issue for this #186 .